core/src/gateway_createfromjson.c

```diff
@@ -129,49 +131,72 @@ GATEWAY_HANDLE Gateway_CreateFromJson(const char* file_path)
     return gw;
 }
 
+static void rollbackModules(size_t success_modules_entries_count, GATEWAY_HANDLE gw, VECTOR_HANDLE modules_added_successfully)
```

Comment:
So I don't need to compute what is the actual module, it was easier to do this with an extra vector.

---
In reply to: [108967297](https://github.com/Azure/azure-iot-gateway-sdk/pull/189#discussion_r108967297) [](ancestors = 108967297)